File: IBNavigationController/Targets/Demos/SubclassObjc/View Controllers/MyCustomNavigationController.m
//
//  MyCustomNavigationController.m
//  NavControllerTest
//
//  Created by Chris Birch on 02/07/2016.
//  Copyright © 2016 Intensebits. All rights reserved.
//

#import "MyCustomNavigationController.h"
@import IBNavigationController;

@interface MyCustomNavigationController ()

@end

@implementation MyCustomNavigationController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do view setup here.
}

-(IBNavigationProxyViewController *)newProxyViewController
{
	return  [[IBNavigationProxyViewController alloc] initWithNibName:@"MyCustomNavProxyViewController" bundle:nil];
}
@end

File: IBNavigationController/Targets/Demos/SubclassSwift/View Controllers/MyCustomNavigationController.swift
//
//  MyCustomNavigationController.swift
//  SubclassSwift
//
//  Created by Chris Birch on 02/07/2016.
//  Copyright © 2016 Intensebits. All rights reserved.
//

import Cocoa
import IBNavigationController
class MyCustomNavigationController: IBNavigationController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do view setup here.
    }
	
	override func newProxyViewController() -> IBNavigationProxyViewController!
	{
		//Here we return our custom subclass of the proxy 
		return  IBNavigationProxyViewController(nibName: "MyCustomNavProxyViewController", bundle: nil)
	}
	
}
